pkg/apis/provisioning/v1alpha1/provisioner.go

```diff
@@ -106,14 +106,22 @@ var (
 	// Reserved labels
 	ProvisionerNameLabelKey      = SchemeGroupVersion.Group + "/name"
 	ProvisionerNamespaceLabelKey = SchemeGroupVersion.Group + "/namespace"
-	ProvisionerUnderutilizedKey  = SchemeGroupVersion.Group + "/underutilized"
-	ProvisionerTTLKey            = SchemeGroupVersion.Group + "/ttl"
+	ProvisionerPhaseLabel        = SchemeGroupVersion.Group + "/lifecycle-phase"
+
+	// Reserved annotations
+	ProvisionerTTLKey = SchemeGroupVersion.Group + "/ttl"
 
 	// Use ProvisionerSpec instead
 	ZoneLabelKey         = "topology.kubernetes.io/zone"
 	InstanceTypeLabelKey = "node.kubernetes.io/instance-type"
 )
 
+const (
+	ProvisionerUnderutilizedPhase = "underutilized"
+	ProvisionerTerminablePhase    = "terminable"
+	ProvisionerDrainingPhase      = "draining"
```

Comment:
What's the difference between Terminable and Draining? Can this be collapsed into a single state: Terminating? That state is managed by the terminator?

Reply:
Terminable is when a node is expired, or after the TTL. Draining is when the node is still in the process of draining. If the underutilization predicate is 0 non-daemonset pods, draining doesn't last very long.

Reply:
There is no difference in amount of API calls done.

Reply:
We should probably chat offline about these terms as it is a bit confusing at the moment (even with the explanation), maybe we can come up with some different words?